netlify-db (#666)
drizzle-team/drizzle-orm-docs: src/content/docs/connect-netlify-db.mdx

import Tab from '@mdx/Tab.astro';
import Tabs from '@mdx/Tabs.astro';
import Npm from "@mdx/Npm.astro";
import Callout from '@mdx/Callout.astro';
import Steps from '@mdx/Steps.astro';
import AnchorCards from '@mdx/AnchorCards.astro';
import Prerequisites from "@mdx/Prerequisites.astro";
import CodeTabs from "@mdx/CodeTabs.astro";
import Section from "@mdx/Section.astro";
import WhatsNextPostgres from "@mdx/WhatsNextPostgres.astro";

> # Drizzle \<\> Netlify DB

Reviewer: ```suggestion
# Drizzle \<\> Netlify Database
```

<Prerequisites>
- Database [connection basics](/docs/connect-overview) with Drizzle

> - Netlify DB - [website](https://docs.netlify.com/build/data-and-storage/netlify-db/)

Reviewer: This is 404ing for now, but will be the docs URL.

```suggestion
- Netlify Database - [website](https://docs.netlify.com/build/data-and-storage/netlify-database/)
```

</Prerequisites>

<Callout type="warning">

> The Netlify DB driver is developed and maintained by the Netlify team.

Reviewer: ```suggestion
The Netlify Database driver is developed and maintained by the Netlify team.
```

> The Netlify DB driver is available in both `drizzle-orm@latest` and `drizzle-orm@beta`. We recommend using the beta tag

Reviewer: Emphasising this a bit more.

```suggestion
The Netlify Database driver is available in both `drizzle-orm@latest` and `drizzle-orm@beta`. We strongly recommend using the beta tag, as it has better support for Netlify's platform-managed migrations
```

</Callout>

> Drizzle has native support for Netlify DB connections, that intelligently selects the optimal underlying Postgres driver based on the runtime environment.

Reviewer: ```suggestion
Drizzle has native support for Netlify Database connections, that intelligently selects the optimal underlying Postgres driver based on the runtime environment.
```

> Netlify DB is a managed Postgres database. On Netlify's platform, the same application code runs in two very different contexts:

Reviewer: ```suggestion
Netlify Database is a managed Postgres database. On Netlify's platform, the same application code runs in two very different contexts:
```

- Serverless functions: no persistent connections, best served by HTTP-based queries
- Server mode: persistent connections via node-postgres

This adapter abstracts that decision away so consumers can write `drizzle()` once and get the right driver automatically.
  
#### Step 1 - Install packages
<Npm>

> drizzle-orm @netlify/db

Reviewer: ```suggestion
drizzle-orm @netlify/database
```

-D drizzle-kit
</Npm>

#### Step 2 - Initialize the driver and make a query
<CodeTabs items={["Zero config", "Connection string", "Explicit client"]}>
```typescript
import { drizzle } from 'drizzle-orm/netlify-db';

> // reads NETLIFY_DB_URL and NETLIFY_DB_DRIVER env vars

Reviewer: ```suggestion
// Connection string is set automatically by the platform
```
